### ISSUE TITLE
STRtree Remove issue.

### ISSUE THREAD

**Issue by ValerioZ** [NetTopologySuite/NetTopologySuite#31]
Personally I found a problem removing element from STRtree.
With this simple code:
>using System;
>using System.Collections.Generic;
>using System.Linq;
>using System.Text;
>using GeoAPI.Geometries;
>using NetTopologySuite.Index.Strtree;

>namespace TestNTSTree
>{
>    class Program
>    {
>        static void Main(string[] args)
>        {
>            STRtree<string> tree = new STRtree<string>();
>            tree.Insert(new Envelope(0, 10, 0, 10), "1");
>            tree.Insert(new Envelope(5, 15, 5, 15), "2");
>            tree.Insert(new Envelope(10, 20, 10, 20), "3");
>            tree.Insert(new Envelope(15, 25, 15, 25), "4");
>            tree.Remove(new Envelope(10, 20, 10, 20), "4");
>        }
>    }
>}

I've an "_{"Object reference not set to an instance of an object."}_" exception on this stacktrace:

>   at NetTopologySuite.Index.Strtree.AbstractSTRtree\`2.Remove(T searchBounds, TItem item)
>   at NetTopologySuite.Index.Strtree.STRtree`1.Remove(Envelope itemEnv, TItem item)
>   at TestNTSTree.Program.Main(String[] args) in >c:\_tfs\AIS\Incubator\TerstDotSpatialTree\TestNTSTree\Program.cs:line 20
>   at System.AppDomain._nExecuteAssembly(RuntimeAssembly assembly, String[] args)
>  at Microsoft.VisualStudio.HostingProcess.HostProc.RunUsersAssembly()
>   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, >ContextCallback callback, Object state, Boolean preserveSyncCtx)
>   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, >ContextCallback callback, Object state, Boolean preserveSyncCtx)
>   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, >ContextCallback callback, Object state)
>   at System.Threading.ThreadHelper.ThreadStart()

I found a really boring issue too: meanwhie you are debugging the code, check the tree imply that the tree is built, hence you cannot insert or remove any item more.

**Comment by FObermaier** [comment]
The "Remove" issue affects JTS also.
The STRtree is a STatic Rtree. So not being able to insert or remove items once the tree was built is a design decision made earlier.

**Comment by airbreather** [comment]
I thought STR stood for Sort-Tile-Recursive, per the [algorithm](https://archive.org/details/nasa_techdoc_19970016975) it uses for packing?

**Comment by FObermaier** [comment]
@airbreather is right, but I am too:
 ```C#
/// A query-only R-tree created using the Sort-Tile-Recursive (STR) algorithm.
/// For two-dimensional spatial data. 
```
See this [link](https://github.com/NetTopologySuite/NetTopologySuite/blob/master/NetTopologySuite/Index/Strtree/STRtree.cs).

**Comment by DGuidi** [comment]
@FObermaier issue can be closed?

**Comment by FObermaier** [comment]
I'd leave it open until jts has [addressed](http://sourceforge.net/p/jts-topo-suite/mailman/message/33260512/) it.

**Comment by DGuidi** [comment]
right move, so we can track also nts integration of jts changes related to this issue.

**Comment by dr-jts** [comment]
JTS is now patched. Thanks for this bug report.

**Comment by FObermaier** [comment]
fixed with 75a0e69c1a8342e75552448c553414fe3148bd24
